### PR TITLE
Move clock function to platform-dependent files

### LINKED ISSUE #35: go get fails on Windows 7 (go version go1.4 windows/amd64)
I couldn't install the library by go getting it. I got the following errors:

C:\Users\hugo>go get github.com/Shopify/go-lua
# github.com/Shopify/go-lua
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:12: undefined: CheckType
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:23: undefined: MetaField
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:24: undefined: CheckType
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:41: undefined: CheckInteger
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:42: undefined: CheckType
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:95: undefined: CheckStackWithMessage
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:101: undefined: Errorf
c:\users\hugo\go\src\github.com\Shopify\go-lua\base.go:326: undefined: SetFunctions
c:\users\hugo\go\src\github.com\Shopify\go-lua\bit32.go:32: undefined: CheckUnsigned
c:\users\hugo\go\src\github.com\Shopify\go-lua\bit32.go:43: undefined: CheckUnsigned
c:\users\hugo\go\src\github.com\Shopify\go-lua\bit32.go:43: too many errors

### PR DESCRIPTION
**Pull request by fbogsany**
Fixes #35

### CODE REVIEW

File: undump_test.go [hunk 1]
```diff
@@ -46,8 +47,8 @@ func TestUndump(t *testing.T) {
 	if err != nil {
 		t.Skipf("testing undump requires luac: %s", err)
 	}
-	source := "lua-tests/checktable.lua"
-	binary := "lua-tests/checktable.bin"
+	source := path.Join("lua-tests", "checktable.lua")
```
Comment (aybabtme) [inline undump_test.go]: `filepath` for files

**Comment by aybabtme** [comment]
:ship:

File: undump_test.go (hunk 1 above)
Comment (aybabtme) [inline undump_test.go]: `filepath` will use `\` slashes in windows, while `path` will use URL style `/`

Comment (fbogsany) [inline undump_test.go]: fixed, thanks @aybabtme !

**Pull request merged**